public/requests.js

```diff
@@ -0,0 +1,122 @@
+        res => get_content('favorites',res.lovedtracks.track)
+    )
+});
+
+/**
+ *
+ * @param method метод Апи (GET или POST)
+ * @param params
+ * @returns {Promise} Выполненный промис
+ */
+async function send_request(method,params) {
+    const url = 'http://ws.audioscrobbler.com/2.0/';
+    const api_key = '906db58ae0258689ba249d53210358ee';
+    try {
+        if (method === 'GET') {
+            const response = await fetch(`${url}?api_key=${api_key}&method=${params.method}${params.param}&format=json`, {
+                method: 'GET'
+            });
+            return await response.json()
+        } else if (method === 'POST') {
+            const sk = 'PEJSZfVNq8UlcfpQoiME1HzKEaFYAClf';
+            const secret = '04e5fe9b4835a8387149c2770345a9af';
+            const url_params = new URLSearchParams({
+                'track': params.track, 'artist': params.artist, 'api_key': api_key,
+                'api_sig': `${md5_encode(api_key, params.artist, sk, params.track, secret)}`, 'sk': sk, 'method': params.method
+            });
+            const response = await fetch(url, {
+                method: 'POST',
+                body: url_params,
+                headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+            })
+        }
+    } catch (err) {
+        alert(`Произошла ошибка ${err.message}`);
+        throw new Error(`Произошла ошибка ${err.message}`);
+    }
+}
+
+//request_method,require_param='',track= '',artist=''
+
+/**
+ *
+ * @param response_content определяет какой контент нужно получить
+ * @param track_array массив треков
+ * @param track название трека
+ * @param artist имя артиста
+ */
+function get_content(response_content,track_array,track=null,artist=null) {
+    const content = document.querySelector('.content');
+    content.innerHTML = "";
+    if (track_array.length) {
+        const button_array = [];
+        track_array.forEach((track) => {
+            const content_favorite = document.createElement("div");
+            const favorite_image = document.createElement("img");
+            const favorite_headline = document.createElement("p");
+            const favorite_quantity = document.createElement("p");
+            content_favorite.className = "content__favorite";
+            favorite_image.className = "favorite__image";
+            favorite_headline.className = "favorite__headline";
+            favorite_quantity.className = "favorite__quantity";
+            favorite_image.src = track.image[3]["#text"];
+            if (response_content === 'search') {
+                favorite_headline.textContent = track.artist;
+            }
+            if (response_content === 'favorites') {
+                const listener = function () {
+                    send_request('POST',{method:'track.unlove',param:'',
+                        track:track.name,artist:track.artist.name});
+                    button_array.forEach(elem => {
```

Comment:
Что-то тут не то
Сейчас после первого нажатия, происходит отписка от нажатия.

Тут нужно сделать по другому.

Нужно сделать так, чтобы перед показом карточек, мы пробегались по всем доступным кнопкам, и отписывались от события.
```suggestion
document.querySelectorAll('.unlove_button').forEach(unlove => {
    unlove.removeEventListener('click', listener)
}); 
```
И далее далее для созданных карточек мы подписываемся на событие.

переменную listener стоит вынести в отдельный метод, который будет брать данные например из атрибута data -track-name и Data-artist-name